File: src/2180.count-integers-with-even-digit-sum.2298/solution.ts
function countEven (num: number): number {
    let result = 0;
    for (let i = 1; i <= num; i++) {
        if (checkEven(i)) {
            result++;
        }
    }
    return result;
}

function checkEven (num:number) {
    let sum = 0;
    while (num) {
        const digit = num % 10;
        sum += digit;
        num = (num - digit) / 10;
    }
    return sum % 2 === 0;
}

File: src/2180.count-integers-with-even-digit-sum.2298/solution.java
class Solution {
    public int countEven(int num) {
        int count = 0;
        while (num>0){
            if(match(num)){
                count++;
            }
            num--;
        }
        return count;
    }

    private boolean match(int num){
        int sum = 0;
        while (num != 0){
            sum += num%10;
            num /= 10;
        }
        return sum%2 == 0;
    }
}